From ef9e0ad7306d8e498527077fd9a68a0216a65121 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:09 -0400
Subject: [tools] include: Define likely/unlikely in linux/compiler.h

Message-id: <1408953210-26343-242-git-send-email-jolsa@redhat.com>
Patchwork-id: 88993
O-Subject: [PATCH RHEL7.1 BZ1133083 241/322] tools include: Define likely/unlikely in linux/compiler.h
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 835d44b9041e578e3e553a57dfffc7003605c93b
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Jan 9 23:00:53 2014 +0900

description
===========
[ Added the new header to tools/perf/MANIFEST ]
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/include/linux/compiler.h b/tools/include/linux/compiler.h
index 0135ccf..fbc6665 100644
--- a/tools/include/linux/compiler.h
+++ b/tools/include/linux/compiler.h
@@ -27,4 +27,12 @@
 # define __weak   __attribute__((weak))
 #endif
 
+#ifndef likely
+# define likely(x)  __builtin_expect(!!(x), 1)
+#endif
+
+#ifndef unlikely
+# define unlikely(x)  __builtin_expect(!!(x), 0)
+#endif
+
 #endif /* _TOOLS_LINUX_COMPILER_H */
diff --git a/tools/perf/MANIFEST b/tools/perf/MANIFEST
index 3170a7f..285f28f 100644
--- a/tools/perf/MANIFEST
+++ b/tools/perf/MANIFEST
@@ -4,6 +4,7 @@ tools/lib/traceevent
 tools/lib/lk
 tools/lib/symbol/kallsyms.c
 tools/lib/symbol/kallsyms.h
+tools/include/linux/compiler.h
 include/linux/const.h
 include/linux/perf_event.h
 include/linux/rbtree.h
-- 
1.7.1